system/hwquirks.c

```diff
@@ -99,6 +102,23 @@ static void amd_k8_revfg_temp(void)
     cpu_temp_offset = 21.0f;
 }
 
+static void loongson_7a00_ehci_workaround(void)
```

Comment:
I don't expect this chipset to be available outside LoongArch computers, so this quirk could be made conditional on `__loongarch_lp64` ?

Reply:
But the LoongArch computers can use other chipsets, for EX: Loongson 3C6000 and later supports the LCL(Loognson coherent link, compatible PCIe4), which supports PCIe4 bridge or chipsset linking, so at this point, I think it is not unique to LoongArch.

Reply:
ACK then.

Reply:
You could make the x86-specific quirks depend on i386 or x86_64, though, if you don't want to clutter LA64 binaries with dead code for x86/x86_64 workarounds.

Reply:
This remark is also valid for A LOT of other code everywhere. I don't think adding a ton of IF is a good idea. At some point, if having unreachable code is really an issue (I don't think so), a separate build is the correct solution.

Reply:
Yes, the architecture-related code has been separated, but some codes still together. I will think about this issue.

Reply:
This workaround is used to turn off EHCI controller of Loongson7A series. Currently, memtest86+ doesn't support usb hot-plugging, and only keyboards. However, the EHCI controller and OHCI controller of Loongson7A seems to  be mutually exclusive, which means if the EHCI controller is probed and started , OHCI will no longer be able to scan any devices, so I turned it off.
@x86fr Aother quesiton, do we need to use the EHCI U-disk? If so, it seems reasonable to detect and start EHCI, if no, why we have to support the EHCI?

Reply:
We support EHCI because some machines don't provide a companion controller and rely on downstream hubs to perform the high speed to low speed translation.

Did you try setting the `USB_IGNORE_EHCI` flag in `usb_init_options` to prevent the EHCI contoller being started? That would be a simpler workaround.